cocos/base/ObjectFactory.cpp

```diff
@@ -104,20 +61,14 @@ void ObjectFactory::destroyInstance()
 
 Ref* ObjectFactory::createObject(const std::string &name)
 {
-    Ref *o = nullptr;
-    do 
-    {
-        const TInfo t = _typeMap[name];
-        if (t._fun != nullptr)
-        {
-            o = t._fun();
-        }else if (t._func != nullptr)
-        {
-            o = t._func();
-        }
-    } while (0);
-   
-    return o;
+    const TInfo t = _typeMap[name];
+    if (t._fun != nullptr) {
+        return t._fun();
+    }
+    else if (t._func != nullptr) {
```

Comment:
It doesn't make sense to have else after return.
https://clang.llvm.org/extra/clang-tidy/checks/readability-else-after-return.html